### ISSUE TITLE
Remove backpassing

### ISSUE THREAD

**Issue by ostcar** [mulias/roc-array2d#8]
The package uses backpassing. When using the package, there are warnings like:

```
── BACKPASSING DEPRECATED in ...a6b24fO1SS_vGNS5MwU-a-3r1niP_7iG6k/Array2D.roc ─

Backpassing (<-) like this will soon be deprecated:

389│          listIndexB <- listIndexOf array.shape indexB |> Result.map
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

You should use a ! for awaiting tasks or a ? for trying results, and
functions everywhere else.
```

**Comment by mulias** [comment]
Oh dang thanks! I'll get this fixed up.

**Comment by mulias** [comment]
Fixed by #9
